File: Backend/KlimabewacherApi/Models/Medicion.cs
using System;
namespace KlimabewacherApi.Properties
{
    public class Medicion
    {
        public int Id { get; set; }
        public string DeviceId { get; set; } = string.Empty;
        public double Temperatura { get; set; }
        public double Humedad { get; set; }
        public string? Ubicacion { get; set; }
        public DateTime FechaHora { get; set; } = DateTime.Now;
    }
}

File: frontend/lib/models/medicion.dart
// models/medicion.dart

class Medicion {
  final String deviceId;
  final double temperatura;
  final double humedad;
  final String ubicacion;
  final DateTime fecha;

  Medicion({
    required this.deviceId,
    required this.temperatura,
    required this.humedad,
    required this.ubicacion,
    required this.fecha,
  });

  factory Medicion.fromJson(Map<String, dynamic> json) {
    return Medicion(
      deviceId: json['deviceId'],
      temperatura: (json['temperatura'] as num).toDouble(),
      humedad: (json['humedad'] as num).toDouble(),
      ubicacion: json['ubicacion'],
      fecha: DateTime.tryParse(json['fechaHora'] ?? '') ?? DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'deviceId': deviceId,
      'temperatura': temperatura,
      'humedad': humedad,
      'ubicacion': ubicacion,
      'fechaHora': fecha.toIso8601String(),
    };
  }
}
